scripts/isolate_tests.py

```diff
@@ -106,13 +84,18 @@ def extract_and_write(f, path):
     write_cases(f, cases)
 
 if __name__ == '__main__':
+    if len(sys.argv) == 1:
+        print("Usage: " + sys.argv[0] + " path-to-file-or-folder-to-extract-code-from [docs]")
+        exit(1)
+
     path = sys.argv[1]
     docs = False
     if len(sys.argv) > 2 and sys.argv[2] == 'docs':
         docs = True
 
     if isfile(path):
-        extract_and_write(path, path)
+        _, tail = split(path)
+        extract_and_write(tail, path)
```

Comment:
```suggestion
        extract_and_write(tail, basename(path))
```
Requires an import of course:
```python
from os.path import basename
```